sapi/phpdbg/config.m4

```diff
@@ -45,13 +45,7 @@ if test "$PHP_PHPDBG" != "no"; then
     AC_MSG_RESULT([disabled])
   fi
 
-  AC_CACHE_CHECK([for userfaultfd faulting on write-protected memory support], ac_cv_phpdbg_userfaultfd_writefault, AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
-    #include <linux/userfaultfd.h>
-    #ifndef UFFDIO_WRITEPROTECT_MODE_WP
-    # error userfaults on write-protected memory not supported
-    #endif
-  ]])], [ac_cv_phpdbg_userfaultfd_writefault=yes], [ac_cv_phpdbg_userfaultfd_writefault=no]))
-  if test "$ac_cv_phpdbg_userfaultfd_writefault" = "yes"; then
+  AC_CHECK_DECL([UFFDIO_WRITEPROTECT_MODE_WP], [
```

Comment:
Does this work for macros? [The manual](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Generic-Declarations.html) for `AC_CHECK_DECL` talks about `symbol (a function, variable, or constant)`.

Reply:
Ah wait, I get it: `This macro actually tests whether it is valid to use symbol as an r-value`. So this should indeed be fine.